cmake/EthCompilerSettings.cmake

```diff
@@ -55,7 +55,7 @@ if (("${CMAKE_CXX_COMPILER_ID}" MATCHES "GNU") OR ("${CMAKE_CXX_COMPILER_ID}" MA
 	add_compile_options(-Wconversion)
 
 	eth_add_cxx_compiler_flag_if_supported(
-		$<$<COMPILE_LANGUAGE:CXX>:-Wextra-semi>
+        $$<$$<COMPILE_LANGUAGE:CXX>:-Wextra-semi>
```

Comment:
This think is CMake Generator Expression and should be resolved during CMake generation phase and never reach the build tool (ninja). The "if supported" check probably does not work in this case because it will be checking the expression verbose (generation phase happens later). Simpler fix can be replacing eth_add_cxx_compiler_flag_if_supported with add_compile_option() and generation expression additionally checking if the compiler is Clang.

Reply:
That's true! ninja should never see any generator expression. That means, that this fix is not really solving the issue. Somehow CMake seem not to evaluate the generator expression correctly, if ninja is used.

Reply:
This should fix it if you want to keep the generic compiler check.
```cmake
    check_cxx_compiler_flag(-Wextra-semi WEXTRA_SEMI)
    if(WEXTRA_SEMI)
        add_compile_options($<$<COMPILE_LANGUAGE:CXX>:-Wextra-semi>)
    endif()

Reply:
@chfast thx! your fix seem to work!

Reply:
For the record, the eth_add_cxx_compiler_flag_if_supported is doing pretty much the same just in this case it would erroneously pass `check_cxx_compiler_flag($<$<COMPILE_LANGUAGE:CXX>:-Wextra-semi>)`.

Reply:
I wonder if we really have to check for C++ here? If not, we could just do
```cmake
eth_add_cxx_compiler_flag_if_supported(-Wextra-semi)
```
I suppose there must be some problem with doing it that way or @axic would have used it already in #10359.

Alternatively, I think this should work:
```cmake
if ($<COMPILE_LANGUAGE:CXX>)
    eth_add_cxx_compiler_flag_if_supported(-Wextra-semi)
endif()
```

In any case, the solution suggested by @chfast is good too, but if possible I think it would be better to have all options go through `eth_add_cxx_compiler_flag_if_supported()` since it does some extra processing.

Reply:
GCC does not like compiling C files with this flag. https://godbolt.org/z/obKK9KT5x.

Reply:
I mean, I expected CMake to not report this flag as supported by the compiler in case of C. But I see where I was wrong no - I just realized it's `check_**cxx**_compiler_flag()`, i.e. we only really check if the C++ compiler supports it and ignore the C compiler.

I see that CMake has `check_c_compiler_flag()` too. In that case wouldn't a more robust solution be to change `eth_add_cxx_compiler_flag_if_supported()` to handle C and C++ separately, each with appropriate function? Actually, its name is a bit misleading because it really adds a flag for both C and C++. So maybe we should also rename it to `eth_add_compiler_flag_if_supported()`?